xrdpkeyb/rdpKeyboard.c

```diff
@@ -668,7 +670,8 @@ rdpkeybControl(DeviceIntPtr device, int what)
     switch (what)
     {
         case DEVICE_INIT:
-            rdpkeybDeviceInit(device, &keySyms, modMap);
+            if (rdpkeybDeviceInit(device, &keySyms, modMap))
+                return BadAlloc;
```

Comment:
I don't like returning BadAlloc here. It created a tight coupling. One function knows why another function fails. Perhaps rdpkeybDeviceInit() itself could return BadAlloc, and its callers could pass it further up.